pkg/api/defaults-custom-cloud-profile.go

```diff
@@ -157,6 +157,22 @@ func (p *Properties) SetAzureStackCloudSpec() error {
 			p.CustomCloudProfile.AzureEnvironmentSpecConfig = &azureStackCloudSpec
 		}
 		AzureCloudSpecEnvMap[AzureStackCloud] = azureStackCloudSpec
+
+		// Update fault domain value to 3 for Azure Stack
+		// on both Master and Agent pools.
+		faultDomaincount := 3
+		if p.MasterProfile != nil && p.MasterProfile.PlatformFaultDomainCount == nil {
+			p.MasterProfile.PlatformFaultDomainCount = &faultDomaincount
```

Comment:
since this change is specific for Azure Stack, please add the condition that only set the value when it is for Azure Stack.
You can use p.IsAzureStackCloud()